hw/ip/rv_dm/data/rv_dm_testplan.hjson

```diff
@@ -432,6 +432,32 @@
       stage: V1
       tests: ["rv_dm_dataaddr_rw_access"]
     }
+    {
+      name: halt_resume
+      desc: '''
+            Verify debug module halt/going/resume behavior.
+            - Program jtag_dmi_ral.dmcontrol.haltreq to 1'b1
+            - After a few usec, acknowledge halt by setting tl_mem_ral.halted to 0. (ack for hart index 0).
+            - Send a single valid command with following process:
+              - Set jtag_dtm_ral.dmi.op to dm::DTM_WRITE
+              - Set jtag_dtm_ral.dmi.address to dm::Command
+              - Set jtag_dtm_ral.dmi.data[31:24] to 0. (Command type AccessRegister).
+            - Check jtag_dmi_ral.abstractcs.busy is 1.
+            - Check flags[0] to make sure internal state machine is at Go state by checking
+              tl_mem_ral.flags[0] is '1'.
+            - Set tl_mem_ral.going to 1. This will make internal state machine move to CmdExecuting state.
```

Comment:
The behavior of this CSR and FSM actually looks to be a bug.

Here's what appears to be happening:

1. Hart jumps to debug_rom `entry` after either the DM requests it (e.g. a halt request) or because the hart executed an `ebreak` instruction.
2. In the debug_rom, the hart will continuously poll the `Flags` CSR for either `go` or `resume`. These are commands sent from the DM to the hart.
    - `go` tells the hart to jump to another place in the debug ROM. The place to `go` to is controlled by the `whereto` CSR--This mapping is probably something else to test.
    - `resume` tells the hart to stop being halted (i.e. to leave Debug Mode and begin executing code normally from where it left off).
3. For the `go` case: The hart _clears_ the `go` flag by writing to the `going` CSR.
    - FYI: `rv_dm` doesn't actually look at the value. It's simply write anything to `going` to clear. Careful with any CSR tests here :wink: 
4. After the hart clears the `go` flag, it runs off to execute the instructions starting from `whereto`.
    - Here is where there may be a bug in the FSM: The halted flag should remain set throughout Debug Mode execution. However, the state would immediately change from `CmdExecuting` to `Idle`, causing a race for the value of `WhereTo`.
5. At the end of any sequence of instructions after `go`, the hart must perform an `ebreak` instruction.
    - This causes the hart to go back to the beginning of the Debug Mode loop, including writing a 1 to `halted` repeatedly.
6. The DM needs to know when the hart is done, when the DM is allowed to forward another command.
    - This could be detected by _the action of writing_ to `halted`, since that could be made to only happen after the `ebreak`
    - However, RV_DM currently uses _the value_ of `halted`. This seems to be a bug.

Reply:
@a-will testplan is updated to check `WhereTo` value for abstract cmd.